dali/python/nvidia/dali/ops.py

```diff
@@ -949,35 +949,46 @@ def current_stream():
         return _CUDAStream(nvidia.dali.python_function_plugin.current_dali_stream())
 
     @staticmethod
-    def function_wrapper_per_sample(function, from_dlpack, to_dlpack, *dlpack_inputs):
+    def check_outputs(outputs, num_outputs):
+        if num_outputs > 1:
+            if not isinstance(outputs, tuple):
+                raise TypeError("The output from a multi-output Python function operator must be a tuple, got: ", type(outputs))
+            if len(outputs) != num_outputs:
+                raise ValueError(f"Unexpected number of outputs from Python function operator - got {len(outputs)}, expected {num_outputs}")
+
+    @staticmethod
+    def function_wrapper_per_sample(function, num_outputs, from_dlpack, to_dlpack, *dlpack_inputs):
         arrays = [from_dlpack(dlpack) for dlpack in dlpack_inputs]
         arr_out = function(*arrays)
         if arr_out is None:
             return
+        PythonFunction.check_outputs(arr_out, num_outputs)
```

Comment:
Judging by the check below, that allowed `list`, aren't we restricting this operator only to return tuples and thus making a breaking change? 
I know that it is not like to happen in production code.

Reply:
If it was ever possible, then it was a bug and likely against the documentation. If the function has multiple outputs, then they should be returned as a tuple, like `return a, b, c` would do.

Reply:
Bugfix it is then.